x-pack/legacy/plugins/code/server/utils/with_request.ts

```diff
@@ -4,24 +4,25 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { RequestFacade } from '../../';
+import { RequestHandlerContext } from 'src/core/server';
 import { AnyObject } from '../lib/esqueue';
 
 export class WithRequest {
   public readonly callCluster: (endpoint: string, clientOptions?: AnyObject) => Promise<any>;
 
-  constructor(readonly req: RequestFacade) {
-    const cluster = req.server.plugins.elasticsearch.getCluster('data');
+  constructor(readonly context: RequestHandlerContext) {
+    // @ts-ignore
+    // const cluster = context.elasticsearch.dataClient.callAsInternalUser;
 
     // @ts-ignore
-    const securityPlugin = req.server.plugins.security;
-    if (securityPlugin) {
-      const useRbac = securityPlugin.authorization.mode.useRbacForRequest(req);
-      if (useRbac) {
-        this.callCluster = cluster.callWithInternalUser;
-        return;
-      }
-    }
-    this.callCluster = cluster.callWithRequest.bind(null, req);
+    // const securityPlugin = req.server.plugins.security;
+    // if (securityPlugin) {
+    //   const useRbac = securityPlugin.authorization.mode.useRbacForRequest(req);
+    //   if (useRbac) {
+    //     this.callCluster = cluster.callWithInternalUser;
+    //     return;
+    //   }
+    // }
+    this.callCluster = context.core.elasticsearch.dataClient.callAsInternalUser;
```

Comment:
the challenging part is we can pass the useRbacForRequest function in here by the `RequestHandlerContext`. However, you can not get a `hapi.Request` instance here.

Reply:
we need to talk to @elastic/kibana-security if they can provide this functionality in NP. otherwise we need to use workaround, for example https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#provide-legacy-platform-api-to-the-new-platform-plugin

Reply:
Assuming the question is whether NP Security plugin exposes `mode.useRbacForRequest`, then the answer is yes. At least in the nearest future you'll be able to get this through security plugin setup contract: `securityPluginSetupContract.authz.mode.useRbacForRequest`.

Reply:
Yes, that's correct. Likely won't be merged until the week before EAH, or right after EAH in the worst case.